### PR TITLE
support activities calling back to scripts

### PR DESCRIPTION
**Pull request by itayd**
this also allow running a specific script function as an activity, as long as it does not try to invoke other activities.

### CODE REVIEW

File: internal/backend/sessions/sessionworkflows/workflow.go [hunk 1]
```diff
@@ -370,8 +374,7 @@ func (w *sessionWorkflow) removeEventSubscription(ctx context.Context, signalID
 }
 
 func (w *sessionWorkflow) run(ctx workflow.Context) (prints []string, err error) {
-	// set before r.Call is called.
-	var callValue sdktypes.Value
+	workflowContextKey := struct{ _ string }{"autokitteh_workflow_context"}
```
Comment (tebeka) [inline internal/backend/sessions/sessionworkflows/workflow.go]: Nitpick. This is not going to guaranteed to be unique in the context:

```go
	s1 := struct{ _ string }{"Hello"}
	s2 := struct{ _ string }{"Hello"}

	fmt.Println(s1 == s2) // true

	m := make(map[any]any)
	m[s1] = 1
	m[s2] = 2
	fmt.Println(m) // map[{}:2]
```

The way to make sure nobody runs over the key is to create a type.
```go
	type ctxKeyType1 int
	var ctxKey1 ctxKeyType1 = 42

	type ctxKeyType2 int
	var ctxKey2 ctxKeyType2 = 42

	m2 := make(map[any]any)
	m2[ctxKey1] = 1
	m2[ctxKey2] = 2
	fmt.Println(m2) // map[42:1 42:2]
```

File: runtimes/starlarkrt/runtime/builtins.go [hunk 2]
```diff
@@ -5,107 +5,181 @@ import (
...
-		}
+	return tlsContext.vctx.ToStarlarkValue(rv)
+}
 
-		if len(args) > 1 {
-			vs, err := kittehs.TransformError(args, vctx.FromStarlarkValue)
-			if err != nil {
-				return nil, fmt.Errorf("cannot convert values from starlark: %w", err)
-			}
+func catchBuiltinFunc(th *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
```
Comment (tebeka) [inline runtimes/starlarkrt/runtime/builtins.go:80]: How is this related to the PR?

File: runtimes/starlarkrt/runtime/builtins.go [hunk 3]
```diff
@@ -5,107 +5,181 @@ import (
...
+	})
+	if err != nil {
+		return nil, fmt.Errorf("cannot convert values from starlark: %w", err)
+	}
+
+	v := sdktypes.NewDictValueFromStringMap(vs)
+	return nil, sdktypes.NewProgramError(v, nil, nil).ToError()
+}
+
+func globalsBuiltinFunc(th *starlark.Thread, bi *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
```
Comment (tebeka) [inline runtimes/starlarkrt/runtime/builtins.go:167]: How is this related to the PR?

File: internal/backend/sessions/sessionworkflows/workflow.go [hunk 4]
```diff
@@ -103,6 +105,8 @@ func (w *sessionWorkflow) addPrint(ctx workflow.Context, print string) {
 func (w *sessionWorkflow) updateState(ctx workflow.Context, state sdktypes.SessionState) {
```
Comment (tebeka) [inline internal/backend/sessions/sessionworkflows/workflow.go:105]: I find it confusion that you call the workflow context `ctx` and there's the built-in context which is traditionally called `ctx`. Maybe rename the parameter to `wctx` ?

**Review by tebeka — APPROVED** [review]
Go go gadget activity.

File: internal/backend/sessions/sessionworkflows/workflow.go (hunk 1 above)
Comment (itayd) [reply internal/backend/sessions/sessionworkflows/workflow.go]: ugh, thanks

File: runtimes/starlarkrt/runtime/builtins.go (hunk 2 above)
Comment (itayd) [reply runtimes/starlarkrt/runtime/builtins.go:80]: did a little refactoring that simplified the structure as part of it.

File: runtimes/starlarkrt/runtime/builtins.go (hunk 3 above)
Comment (itayd) [reply runtimes/starlarkrt/runtime/builtins.go:167]: ditto

File: internal/backend/sessions/sessionworkflows/workflow.go (hunk 4 above)
Comment (itayd) [reply internal/backend/sessions/sessionworkflows/workflow.go:105]: done. i think from now on for workflow context we'll use wctx
